app/models/host/discovered.rb

```diff
@@ -117,9 +117,11 @@ def populate_fields_from_facts(facts = self.facts_hash, ignored_type = nil)
     if Setting::Discovered.discovery_lock? && self.subnet.tftp?
       begin
         @host = self
-        Rails.logger.info "Locking discovered host #{self.mac} in subnet #{subnet}"
-        template = unattended_render(::ProvisioningTemplate.find_by_name(Setting['discovery_lock_template']).template)
-        subnet.tftp_proxy.set mac, :pxeconfig => template
+        TemplateKind::PXE.each do |kind|
+          Rails.logger.info "Locking discovered host #{self.mac} in subnet #{subnet} via #{kind} template"
+          template = unattended_render(::ProvisioningTemplate.find_by_name(Setting["discovery_#{kind}_lock_template"]).template)
```

Comment:
Stubbing is fine too, probably better as adding fixtures from rails engines is cumbersome